packages/react-composites/src/composites/CallComposite/pages/ConfigurationPage.tsx

```diff
@@ -63,6 +63,12 @@ export const ConfigurationPage = (props: ConfigurationPageProps): JSX.Element =>
   /* @conditional-compile-remove(rooms) */
   const rolePermissions = _usePermissions();
   /* @conditional-compile-remove(rooms) */
+  if (!rolePermissions.cameraButton) {
```

Comment:
This splits the business logic of what errors to show between the error bar selector and the configuration page UI component.

The right approach here is to plumb the role from stateful layer into the error bar selector and act on it there.

I understand that for the first public preview, role is provided as a composite argument, so it isn't available to the selector. But this is a very short-term stop-gap.

Please add a TODO comment here to move this logic to the error bar selector once role is plubmed in from headless calling SDK, and create a task for the same under an appropriate user story.

Reply:
(somewhat tangential, we need to know the role in the configuration page (i.e. before a call is joined), which sounds like it won't be possible to come from stateful client? One possibility would be to pass into the adapter ctor though and use in the adapter selectors)

Reply:
Ok I will add a TODO comment when the role is available in the stateful client. 

That's a good thought on the availability of the role in the configuration page. I will have to give it some thought when we remove role from the CallComposite props.